p2p/channel.go

```diff
@@ -380,17 +554,18 @@ func newBlockCrypt(privateKey PrivateKey, peerPublicKey PublicKey) (kcp.BlockCry
 	return blockCrypt, nil
 }
 
-func listenUDP(punchedConn *net.UDPConn) (*net.UDPConn, error) {
-	punchedConn.Close()
-
-	raddr := punchedConn.RemoteAddr().(*net.UDPAddr)
-	network := "udp4"
-	if raddr.IP.To4() == nil {
-		network = "udp"
+func errNetClose(err error) bool {
+	// Hack. See https://github.com/golang/go/issues/4373 which should expose net close error with 1.15.
+	if strings.Contains(err.Error(), "use of closed network connection") {
```

Comment:
> S1008: should use 'return strings.Contains(err.Error(), "use of closed network connection")' instead of 'if strings.Contains(err.Error(), "use of closed network connection") { return true }; return false' (gosimple)